repose-aggregator/components/filters/scripting-filter/src/main/scala/org/openrepose/filters/scripting/ScriptingFilter.scala

```diff
@@ -0,0 +1,139 @@
+    configurationService.subscribeTo(
+      filterConfig.getFilterName,
+      configurationFile,
+      xsdURL,
+      this,
+      classOf[ScriptingConfig])
+  }
+
+  override def destroy(): Unit = {
+    configurationService.unsubscribeFrom(configurationFile, this)
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    if (!isInitialized) {
+      logger.warn("Scripting filter has not yet initialized")
+      servletResponse.asInstanceOf[HttpServletResponse].sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE)
+    } else {
+      val wrappedRequest = new HttpServletRequestWrapper(servletRequest.asInstanceOf[HttpServletRequest])
+      val wrappedResponse = new HttpServletResponseWrapper(servletResponse.asInstanceOf[HttpServletResponse],
+        ResponseMode.MUTABLE,
+        ResponseMode.MUTABLE)
+
+      val bindings = new SimpleBindings()
+      bindings.put("request", wrappedRequest)
+      bindings.put("response", wrappedResponse)
+      bindings.put("filterChain", filterChain)
+
+      logger.debug("Running script")
+      Try(scriptRunner.run(bindings)) match {
+        case Success(_) =>
+          wrappedResponse.commitToResponse()
+        case Failure(e) =>
+          logger.error("Processing failure -- the script threw an Exception", e)
+          servletResponse.asInstanceOf[HttpServletResponse].sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR)
+      }
+    }
+  }
+
+  override def isInitialized: Boolean = initialized
+
+  override def configurationUpdated(configurationObject: ScriptingConfig): Unit = {
+    val scriptEngineManager = new ScriptEngineManager()
+
+    scriptRunner = Option(scriptEngineManager.getEngineByName(configurationObject.getScript.getLanguage)) match {
+      case Some(engine) =>
+        ScriptRunner(configurationObject.getScript.getValue, engine)
+      case None =>
+        logger.error(configurationObject.getScript.getLanguage + " is not a supported language")
+        throw new UpdateFailedException(configurationObject.getScript.getLanguage + " is not a supported language", null)
+    }
+
+    initialized = true
+  }
+
+  private sealed trait ScriptRunner {
+    def run(bindings: Bindings): AnyRef
+  }
+
+  private class CompiledScriptRunner(compiledScript: CompiledScript) extends ScriptRunner {
+    override def run(bindings: Bindings): AnyRef = {
+      compiledScript.eval(bindings)
+    }
+  }
+
+  private class StringScriptRunner(script: String, scriptEngine: ScriptEngine) extends ScriptRunner {
+    override def run(bindings: Bindings): AnyRef = {
+      scriptEngine.eval(script, bindings)
+    }
+  }
+
+  private object ScriptRunner {
+    def apply(script: String, engine: ScriptEngine): ScriptRunner = {
+      engine match {
+        case c: Compilable =>
+          logger.debug("Creating compiled script runner")
+          new CompiledScriptRunner(c.compile(script))
```

Comment:
It was. CompiledScriptRunner is not a case class because I didn't need to give access to the constructor parameter. Since it isn't a case class, it isn't provided an "apply" function by default. The apply function is what allows you to neglect using the "new" keyword.

I suppose I could have made it a case class, but I don't see any reason to.